server/auth/azure.js

```diff
@@ -0,0 +1,88 @@
+// @flow
+import passport from "@outlinewiki/koa-passport";
+import jwt from "jsonwebtoken";
+import Router from "koa-router";
+import { capitalize } from "lodash";
+import { Strategy as AzureStrategy } from "passport-azure-ad-oauth2";
+import accountProvisioner from "../commands/accountProvisioner";
+import env from "../env";
+import auth from "../middlewares/authentication";
+import passportMiddleware from "../middlewares/passport";
+import { StateStore } from "../utils/passport";
+
+const router = new Router();
+const providerName = "azure";
+const AZURE_CLIENT_ID = process.env.AZURE_CLIENT_ID;
+const AZURE_CLIENT_SECRET = process.env.AZURE_CLIENT_SECRET;
+
+const scopes = [];
+
+export const config = {
+  name: "Office 365",
```

Comment:
"Microsoft", "Azure", "Teams", "Office 365" – I'm tempted to just do "Microsoft", it seems to cover all the possible options.